dimagi_lake/aggregation/aggregation_helpers/agg_flwc_administration.py

```diff
@@ -0,0 +1,138 @@
+from datetime import datetime
+
+from dimagi_lake.aggregation.aggregation_helpers.base_helper import \
+    BaseAggregationHelper
+from dimagi_lake.utils import clean_name
+from spark_session_handler import SPARK
+
+
+class AggFlwcAdministartionHelper(BaseAggregationHelper):
+
+
+    @property
+    def source_tablename(self):
+        return "flwc_management_flwc_administration"
+
+    def aggregate(self):
+        df = self.pre_process()
+        df.show()
+        df = self.apply_transformation(df)
+
+        return self.enforce_schema(df)
+
+    @property
+    def columns(self):
+        return {
+            "infrastructure_available": (
+                'form',
+                'flwc_information_and_infrastructure_group',
+                'infrastructure_at_flwc_details_group',
+                'infrastructure_available'
+            ),
+            "gme_available": (
+                "form",
+                "essentials_at_flwc_details_group",
+                "growth_measuring_equipments_details_group",
+                "gme_available"
+            ),
+            "utility_items_list": (
+                "form",
+                "essentials_at_flwc_details_group",
+                "other_utility_items_details_group",
+                "utility_items_list"
+            ),
+            "timeend": (
+                "form",
+                "meta",
+                "timeEnd"
+            )
+        }
+
+    def pre_process(self):
+        df = SPARK.sql(f"""
+        SELECT
+            flwc_id,
+            supervisor_id,
+            project_id,
+            district_id,
+            state_id,
+            {self.get_column('timeend')} as timeend,
+            int(array_contains(split({self.get_column('infrastructure_available')}, ' '), 'safe_drinking_water')) as safe_drinking_water,
+            int(array_contains(split({self.get_column('infrastructure_available')}, ' '), 'toilet')) as functional_toilet,
+            int(array_contains(split({self.get_column('gme_available')}, ' '), 'weighing_scale_infants'))  as weighing_scale_infants,
+            int(array_contains(split({self.get_column('gme_available')}, ' '), 'weighing_scale_mother_and_child'))  as weighing_scale_mother_and_child,
+            int(array_contains(split({self.get_column('gme_available')}, ' '), 'infantometer'))  as infantometer,
+            int(array_contains(split({self.get_column('gme_available')}, ' '), 'stadiometer'))  as stadiometer,
+            int(array_contains(split({self.get_column('utility_items_list')}, ' '), 'medicine_kit'))  as medicine_kit
+        FROM {self.database_name}.{self.source_tablename} as source_table
+        WHERE add_months(month, 6) > '{self.month}' AND month < add_months('{self.month}', 1)
+        """)
+        return df
+
+    def apply_transformation(self, df):
+
+        df.createOrReplaceTempView("pass1")
```

Comment:
where does "pass1" come from?

Reply:
I named it as the output from the `pre_process` layer after 1st pass.
I think I would name it a bit better.